src/Cardano/Address/Style/Shelley.hs

```diff
@@ -0,0 +1,217 @@
+                seed
+
+instance HardDerivation Shelley where
+    type AccountIndexDerivationType Shelley = 'Hardened
+    type AddressIndexDerivationType Shelley = 'Soft
+    type WithAccountStyle Shelley = AccountingStyle
+
+    deriveAccountPrivateKey (Shelley rootXPrv) accIx =
+        let
+            purposeIx =
+                toEnum @(Index 'Hardened _) $ fromEnum purposeIndex
+            coinTypeIx =
+                toEnum @(Index 'Hardened _) $ fromEnum coinTypeIndex
+            purposeXPrv = -- lvl1 derivation; hardened derivation of purpose'
+                deriveXPrv DerivationScheme2 rootXPrv purposeIx
+            coinTypeXPrv = -- lvl2 derivation; hardened derivation of coin_type'
+                deriveXPrv DerivationScheme2 purposeXPrv coinTypeIx
+            acctXPrv = -- lvl3 derivation; hardened derivation of account' index
+                deriveXPrv DerivationScheme2 coinTypeXPrv accIx
+        in
+            Shelley acctXPrv
+
+    deriveAddressPrivateKey (Shelley accXPrv) accountingStyle addrIx =
+        let
+            changeCode =
+                toEnum @(Index 'Soft _) $ fromEnum accountingStyle
+            changeXPrv = -- lvl4 derivation; soft derivation of change chain
+                deriveXPrv DerivationScheme2 accXPrv changeCode
+            addrXPrv = -- lvl5 derivation; soft derivation of address index
+                deriveXPrv DerivationScheme2 changeXPrv addrIx
+        in
+            Shelley addrXPrv
+
+instance SoftDerivation Shelley where
+    deriveAddressPublicKey (Shelley accXPub) accountingStyle addrIx =
+        fromMaybe errWrongIndex $ do
+            let changeCode = toEnum @(Index 'Soft _) $ fromEnum accountingStyle
+            changeXPub <- -- lvl4 derivation in bip44 is derivation of change chain
+                deriveXPub DerivationScheme2 accXPub changeCode
+            addrXPub <- -- lvl5 derivation in bip44 is derivation of address chain
+                deriveXPub DerivationScheme2 changeXPub addrIx
+            return $ Shelley addrXPub
+      where
+        errWrongIndex = error $
+            "deriveAddressPublicKey failed: was given an hardened (or too big) \
+            \index for soft path derivation ( " ++ show addrIx ++ "). This is \
+            \either a programmer error, or, we may have reached the maximum \
+            \number of addresses for a given wallet."
+
+{-------------------------------------------------------------------------------
+                                 Key generation
+-------------------------------------------------------------------------------}
+
+-- | Size, in bytes, of a public key (without chain code)
+publicKeySize :: Int
+publicKeySize = 32
+
+-- Serialized length in bytes of a Single Address
+addrSingleSize :: Int
+addrSingleSize = 1 + publicKeySize
+
+-- Serialized length in bytes of a Grouped Address
+addrGroupedSize :: Int
+addrGroupedSize = addrSingleSize + publicKeySize
```

Comment:
Hmmm. Grouped and Single addresses were Jörmungandr constructs. 

Note that, what is currently called "Shelley" in cardano-wallet should really be called "Jormungandr" ! 
Shelley addresses are actually going to be different. What should land in this module should be the Shelley stuff, not the Jörmungandr ones.